lisa/runners/lisa_runner.py

```diff
@@ -703,7 +708,11 @@ def _create_platform_requirement(self) -> Optional[schema.NodeSpace]:
         # fill in required fields as max capability. So it can be
         # used as a capability in next steps to merge with test requirement.
         if not platform_requirement.disk:
-            platform_requirement.disk = schema.DiskOptionSettings()
+            if self.platform.type_name() == constants.PLATFORM_AZURE:
```

Comment:
We shouldn't create a reference from generic code to platform specified. The AzureDiskOptionSettings() should be overridden in azure platform, not in runner.

Reply:
Any suggestions on how to handle this? I need this to be platform specific otherwise `intersect` doesn't work

Reply:
It should be handled in this method. What exact "doesn't work" mean here? We can discuss it tomorrow.
https://github.com/microsoft/lisa/blob/main/lisa/sut_orchestrator/azure/platform_.py#L2071